subgraph/src/utils/entity.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-underscore-dangle */
+import { BigInt as GraphBN, Bytes, ethereum } from "@graphprotocol/graph-ts";
+
+import { Account, MACI, User } from "../../generated/schema";
+
+import { DEFAULT_MACI_ID } from "./constants";
+
+export const createOrLoadMACI = (event: ethereum.Event, stateTreeDepth: GraphBN = GraphBN.fromI32(10)): MACI => {
+  let maci = MACI.load(DEFAULT_MACI_ID);
+
+  if (!maci) {
+    maci = new MACI(DEFAULT_MACI_ID);
+    maci.stateTreeDepth = stateTreeDepth;
+    maci.updatedAt = event.block.timestamp;
+    maci.blockNumber = event.block.number;
+    maci.txHash = event.transaction.hash;
+
+    maci.numPoll = GraphBN.zero();
+    maci.numSignUps = GraphBN.zero();
+    maci.latestPoll = Bytes.empty();
+    maci.save();
+  }
+
+  return maci;
+};
+
+export const createOrLoadUser = (publicKeyX: GraphBN, publicKeyY: GraphBN, event: ethereum.Event): User => {
+  const publicKey = `${publicKeyX.toString()} ${publicKeyY.toString()}`;
+  let user = User.load(publicKey);
+
+  if (!user) {
+    user = new User(publicKey);
+    user.timestamp = event.block.timestamp;
+    user.blockNumber = event.block.number;
+    user.txHash = event.transaction.hash;
+    user.save();
+  }
+
+  return user;
+};
+
+export const createOrLoadAccount = (
```

Comment:
What's the difference with the user?

Reply:
User has polls (for coordinator) and accounts (for voters) while account only has state index, voice credits and etc.
This is not a new code, all these mapper changes are from https://github.com/privacy-scaling-explorations/maci/pull/1203